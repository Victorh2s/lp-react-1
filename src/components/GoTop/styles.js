import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: none;
    color: ${theme.colors.colorbuttons};
    border: 0.1rem solid ${theme.colors.colorbuttons};
    border-radius: 5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
  `}
`;
