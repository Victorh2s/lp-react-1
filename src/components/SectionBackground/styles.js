import styled, { css } from 'styled-components';

const backgroundActive = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
${({ theme, background }) => css`
transition: all 300ms ease-in-out;
background: ${theme.colors.white};
color: ${theme.colors.primaryColor};
${background && backgroundActive(theme)}
min-height: 80vh;
display: flex;
align-items: center;
`}
`;
