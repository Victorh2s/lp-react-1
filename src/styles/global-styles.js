import { createGlobalStyle, css } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=Open+Sans:ital,wght@0,400;0,600;0,800;1,700&display=swap');
  * {
    margin:0;
    padding:0;
  }
  html {
    font-size: 62.5%;
  }
  body {
    font-size:2.4rem;
    font-family: ${({theme})=> theme.font,family.default};
  }
  h1 {
    font-size: 2.4rem;
    font-family:  ${({theme})=> theme.font,family.secondary};
  }
`;
