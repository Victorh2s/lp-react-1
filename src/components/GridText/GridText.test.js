import { screen } from '@testing-library/react';
import { GridText } from '.';
import { renderTheme } from '../../styles/render-theme';
describe('<GridText/>', () => {
  it('should render', () => {
    renderTheme(<GridText>Children</GridText>);
    const gridText = screen.getByRole('heading');
    expect(gridText).toBeInTheDocument();
  });
});
