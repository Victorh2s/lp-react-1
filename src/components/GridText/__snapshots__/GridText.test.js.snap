// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridText/> should render GridText with background 1`] = `
<div>
  <div
    class="sc-gKXOVf joDErG"
    id=""
  >
    <div
      class="sc-jSMfEi dRbZhr"
    >
      <div
        class="sc-dkzDqf hsxRpz"
      >
        <h2
          class="sc-bczRLJ hGDuJC"
        >
          My grid
        </h2>
        <div
          class="sc-gsnTZi gNZDCw"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </div>
        <div
          class="sc-hKMtZM lkscSh"
        >
          <div
            class="sc-eCYdqJ lluHNs"
          >
            <h3
              class="sc-bczRLJ fnHBwd"
            >
              Teste 1
            </h3>
            <div
              class="sc-gsnTZi gNZDCw"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </div>
          </div>
          <div
            class="sc-eCYdqJ lluHNs"
          >
            <h3
              class="sc-bczRLJ fnHBwd"
            >
              Teste 2
            </h3>
            <div
              class="sc-gsnTZi gNZDCw"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </div>
          </div>
          <div
            class="sc-eCYdqJ lluHNs"
          >
            <h3
              class="sc-bczRLJ fnHBwd"
            >
              Teste 3
            </h3>
            <div
              class="sc-gsnTZi gNZDCw"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridText/> should render GridText without background 1`] = `
<div>
  <div
    class="sc-gKXOVf joDErG"
    id=""
  >
    <div
      class="sc-jSMfEi dRbZhr"
    >
      <div
        class="sc-dkzDqf hsxRpz"
      >
        <h2
          class="sc-bczRLJ hGDuJC"
        >
          My grid
        </h2>
        <div
          class="sc-gsnTZi gNZDCw"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </div>
        <div
          class="sc-hKMtZM lkscSh"
        >
          <div
            class="sc-eCYdqJ lluHNs"
          >
            <h3
              class="sc-bczRLJ fnHBwd"
            >
              Teste 1
            </h3>
            <div
              class="sc-gsnTZi gNZDCw"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </div>
          </div>
          <div
            class="sc-eCYdqJ lluHNs"
          >
            <h3
              class="sc-bczRLJ fnHBwd"
            >
              Teste 2
            </h3>
            <div
              class="sc-gsnTZi gNZDCw"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </div>
          </div>
          <div
            class="sc-eCYdqJ lluHNs"
          >
            <h3
              class="sc-bczRLJ fnHBwd"
            >
              Teste 3
            </h3>
            <div
              class="sc-gsnTZi gNZDCw"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
